### ISSUE TITLE
WP_Async_Request causing fatal error with Gravity Forms

### ISSUE THREAD

**Issue by apos37** [awesomemotive/WP-Mail-SMTP#143]
## Expected Behavior
I should be able to fill out a form with email notifications without getting a fatal error.

## Current Behavior
I get the following fatal error:

[24-Apr-2025 16:30:14 UTC] PHP Fatal error:  Uncaught Error: Call to undefined method GF_Feed_Processor::maybe_wp_die() in /home/xxx/public_html/wp-content/plugins/gravityforms/includes/libraries/gf-background-process.php:945
Stack trace:
#0 /home/xxx/public_html/wp-content/plugins/gravityforms/includes/libraries/gf-background-process.php(553): GF_Background_Process->handle()
#1 /home/xxx/public_html/wp-includes/class-wp-hook.php(324): GF_Background_Process->maybe_handle()
#2 /home/xxx/public_html/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()
#3 /home/xxx/public_html/wp-includes/plugin.php(517): WP_Hook->do_action()
#4 /home/xxx/public_html/wp-admin/admin-ajax.php(192): do_action()
#5 {main}
  thrown in /home/xxx/public_html/wp-content/plugins/gravityforms/includes/libraries/gf-background-process.php on line 945

This is happening on some forms that email notifications, but not all, so I haven't narrowed down what is different. All I can say is that it is happening across several of the sites I'm managing and it just started after Gravity Forms' latest update today.

## Possible Solution
I obviously contacted Gravity Forms support first since the error is coming from their plugin; however, they responded saying, "The `maybe_wp_die` method is defined in the WP_Async_Request class which the GF_Feed_Processor and GF_Background_Process classes extend. Best guess is that another plugin is including an older version of the WP_Async_Request class." 

In doing a string search across my theme and plugins I found that the WP Mail SMTP plugin is the only other source using `WP_Async_Request`. I then tested deactivating the WP Mail SMTP and completing my form again and I no longer get errors.

**Comment by alexander-sakal** [comment]
Hey @apos37. Very interesting issue. I found that the [Action Scheduler library](https://github.com/woocommerce/action-scheduler), which is bundled with the WP Mail SMTP plugin, uses an outdated version of the [WP Background Processing](https://github.com/deliciousbrains/wp-background-processing) library, and that’s what’s causing the issue. Unfortunately, we can’t fix this from our end. I believe it would be beneficial for Gravity Forms to address the issue on their side, since the Action Scheduler library is widely used and bundled with many major plugins.

I'll also open a GitHub issue in the Action Scheduler repository.

**Comment by alexander-sakal** [comment]
Here is GitHub issue in the Action Scheduler repository: https://github.com/woocommerce/action-scheduler/issues/1267

**Comment by jimtrue** [comment]
We have released Gravity Forms 2.9.7.2 to address the issue in Gravity Forms. The release is on Autoupdate, but can also be downloaded through the regular methods for manual install.

**Comment by capuderg** [comment]
Thank you @jimtrue!
